File: ai-wellness-service/src/main/java/com/kovanlabs/wellness/service/LeaderboardService.java
package com.kovanlabs.wellness.service;

import com.kovanlabs.wellness.dto.team.TeamLeaderboardResponse;

import java.time.Instant;

public interface LeaderboardService {

    TeamLeaderboardResponse getTeamLeaderboard(Long teamId, Instant startTime, Instant endTime);
}

File: web-dashboard/src/api/leaderboardService.ts
import { apiClient } from './apiClient';
import { TeamLeaderboardResponse } from '../types';

export const leaderboardService = {
  getTeamLeaderboard: async (teamId: number, startTime?: string, endTime?: string): Promise<TeamLeaderboardResponse> => {
    const response = await apiClient.get<TeamLeaderboardResponse>(`/api/teams/${teamId}/leaderboard`, {
      params: { startTime, endTime },
    });
    return response.data;
  },
};
